docs/development/benchmark_with_djl.md

```diff
@@ -0,0 +1,205 @@
+```
+
+### Benchmark from ModelZoo
+
+#### MXNet
+
+Resnet50 image classification model:
+
+```
+./gradlew :examples:benchmark --args="-c 1 -s 1,3,224,224 -n ai.djl.mxnet:resnet:0.0.1 -r {'layers':'50','flavor':'v2','dataset':'imagenet'}"
+```
+
+#### PyTorch
+
+SSD object detection model:
+
+```
+./gradlew :examples:benchmark -Dai.djl.default_engine=PyTorch --args="-c 1 -s 1,3,300,300 -n ai.djl.pytorch:ssd -r {'size':'300','backbone':'resnet50'}"
+```
+
+
+## Configuration of Benchmark script
+
+To start your benchmarking, we need to make sure we provide the following information.
+
+- The Deep Learning Engine
+- The source of the model
+- How many runs you would like to make
+- Sample input for the model
+- (Optional) Multi-thread benchmark
+
+The benchmark script located [here](https://github.com/awslabs/djl/blob/master/examples/src/main/java/ai/djl/examples/inference/benchmark/Benchmark.java).
+
+Firstly create a terminal/cmd and `cd` to the `examples` folder and do the following:
+
+```
+./gradlew benchmark --args="--help"
+```
+
+or Windows:
+
+```
+..\gradlew.bat benchmark --args="--help"
+```
+
+This will print out the possible arguments to pass in
+
+### Step 1: Pick your deep engine
+
+By default, the above script will use MXNet as the default Engine, but you can always change that by adding the followings:
+
+```
+-Dai.djl.default_engine=TensorFlow # tensorflow
+-Dai.djl.default_engine=PyTorch # pytorch
+```
+to change your default engine.
+
+### Step 2: Identify the source of your model
+
+DJL accept variety of models came from different places.
+
+#### Remote location
+
+The following is a pytorch model
+
+```
+-Dai.djl.repository.zoo.location=https://alpha-djl-demos.s3.amazonaws.com/model/djl-blockrunner/pytorch_resnet18.zip?model_name=traced_resnet18
+```
+We would recommend to make model files in a zip for better file tracking.
+
+#### Local directory
+
+Mac/Linux
+
+```
+-Dai.djl.repository.zoo.location=file:///pytorch_resnet18.zip?model_name=traced_resnet18
+```
+
+Windows
+
+```
+-Dai.djl.repository.zoo.location=file:///C:/pytorch_resnet18.zip?model_name=traced_resnet18 # windows
+```
+
+#### DJL Model zoo
+
+You can run `listmodels` to list available models that you can use from different model zoos.
+
+```
+./gradlew listmodels # MXNet models
+./gradlew listmodels -Dai.djl.default_engine=TensorFlow # TensorFlow models
+./gradlew listmodels -Dai.djl.default_engine=PyTorch # PyTorch models
+```
+
+After that, just simply copy the json formatted criteria like `{"layers":"18","flavor":"v1","dataset":"imagenet"}` with the model name like `ai.djl.mxnet:resnet:0.0.1`.
+Then, you can just pass These information in the `--args`:
+
+```
+-n ai.djl.mxnet:resnet:0.0.1 -r {"layers":"18","flavor":"v1","dataset":"imagenet"}
```

Comment:
```suggestion
-n ai.djl.mxnet:resnet -r {"layers":"18","flavor":"v1","dataset":"imagenet"}
```

Reply:
This is what user see when they do `listmodels`. I suggest keep it consistent to avoid confusion

Reply:
> This is what user see when they do `listmodels`. I suggest keep it consistent to avoid confusion

We don't accept version selection currently in model zoo. ModelZoo hard-coded supported version. For example 0.8.0 ModelZoo only works with 0.0.1 but 0.9.0 model zoo requires 0.0.2.